app/controllers/api/base_controller/parser.rb

```diff
@@ -72,16 +72,20 @@ def validate_api_parameters
       def parse_href(href)
         if href
           path = href.match(/^http/) ? URI.parse(href).path.sub(/\/*$/, '') : href
-          path = "#{@prefix}/#{path}" unless path.match(@prefix)
+          path = "#{prefix}/#{path}" unless path.match(prefix)
           path = path.sub(/\/*$/, '')
           return href_collection_id(path)
         end
         [nil, nil]
       end
 
+      def prefix
+        @prefix ||= "/#{Settings.base.module}"
+      end
+
       def href_collection_id(path)
         path_array = path.split('/')
```

Comment:
I know this is original code, but can this be changed to instead decompose into named variables?  `path_array[2]` is meaningless, but naming what it is immediately on splitting will be helpful.